gateway-sp-comms/src/recv_handler/mod.rs

```diff
@@ -207,22 +309,196 @@ impl RecvHandler {
             packet.data.as_ptr() as usize as u64,
             u64::from(packet.len)
         ));
-        debug!(
-            &self.log,
-            "received serial console data from {:?}: {:?}", port, packet
-        );
-        self.sp_state(port)
-            .serial_console
-            .lock()
-            .unwrap()
-            .push(packet, &self.log);
+
+        let tasks = self.sp_state(port).serial_console_tasks.lock().unwrap();
+
+        // if we have a task managing an active client connection, forward this
+        // packet to it; otherwise, drop it on the floor.
```

Comment:
Is there anything else we might want to do that could be useful such as using a fixed size ring buffer to let recently attached clients get some fixed history or increment a counter (although I'm not sure what we'd do with "you missed 1000 bytes, sorry"

Reply:
Maybe? This PR actually deleted `serial_console_history.rs` which had a ringbuffer of the most recent N packets from each SP. We had to have something like that with the non-websocket API since it was a polling interface, but it was starting to make me nervous that we might be sitting around with potentially-sensitive strings in memory for an arbitrarily long time. I could bring it (or something similar to it) back though. Are you worried about the contents of those ringbuffers?